inc/Engine/CriticalPath/APIClient.php

```diff
@@ -0,0 +1,234 @@
+<?php
+namespace WP_Rocket\Engine\CriticalPath;
+
+use WP_Error;
+
+/**
+ * Class APIClient
+ *
+ * @package WP_Rocket\Engine\CriticalPath
+ */
+class APIClient {
+
+	/**
+	 * Constant url for Critical Path API job.
+	 */
+	const API_URL = 'https://cpcss.wp-rocket.me/api/job/';
+
+	/**
+	 * Sends a generation request to the Critical Path API
+	 *
+	 * @since 3.6
+	 *
+	 * @param string $post_url The post URL.
+	 * @return array
+	 */
+	public function send_generation_request( $post_url ) {
+		$response = wp_remote_post(
+			self::API_URL,
+			[
+				// This filter is documented in inc/Engine/CriticalPath/CriticalCSSGeneration.php.
+				'body' => apply_filters(
+					'rocket_cpcss_job_request',
+					[
+						'url' => $post_url,
+					]
+				),
+			]
+		);
+
+		return $this->prepare_response( $response, $post_url );
+	}
+
+	/**
+	 * Prepare the response to be returned.
+	 *
+	 * @param array|WP_Error $response The response or WP_Error on failure.
+	 * @param string         $post_url Url for the post to be checked.
+	 * @return array|WP_Error
+	 * @since 3.6
+	 */
+	private function prepare_response( $response, $post_url ) {
+		$response_data        = $this->get_response_data( $response );
+		$response_status_code = $this->get_response_status( $response );
+		$succeeded            = $this->get_response_success( $response_status_code, $response_data );
+
+		if ( $succeeded ) {
+			return $response_data;
+		}else {
+			$response_code    = $this->get_response_code( $response );
+			$response_message = $this->get_response_message( $response_status_code, $response_data, $post_url );
+
+			if ( 200 === $response_status_code ) {
+				$response_status_code = 400;
+			}
+
+			return new WP_Error(
+				$response_code,
+				$response_message,
+				[
+					'status' => $response_status_code,
+				]
+			);
+		}
+	}
+
+	/**
+	 * Get the status of response.
+	 *
+	 * @param int   $response_code Response code to check success or failure.
+	 * @param array $response_data Array of data returned from request.
+	 * @return bool success or failed.
+	 * @since 3.6
+	 */
+	private function get_response_success( $response_code, $response_data ) {
+		return (
+			200 === $response_code &&
+			! empty( $response_data ) &&
+			(
+				( isset( $response_data->status ) && 200 === $response_data->status ) ||
+				( isset( $response_data->data ) && isset( $response_data->data->id ) )
+			)
+		);
+	}
+
+	/**
+	 * Get response status code/number.
+	 *
+	 * @since 3.6
+	 *
+	 * @param array|WP_Error $response The response or WP_Error on failure.
+	 * @param null|int       $status Status code to overwrite the response status.
+	 * @return int|string status code|number of response.
+	 */
+	private function get_response_status( $response, $status = null ) {
+		if ( is_null( $status ) ) {
+			$status = wp_remote_retrieve_response_code( $response );
+		}
+
+		return $status;
```

Comment:
```suggestion
		return (int) $status;
```

Let's typehint the return to always return an `int`

Reply:
Ok but this method wp_remote_retrieve_response_code returns either the number of status code or empty string and phpdoc for it contains that it returns 
```
@return int|string The response code as an integer. Empty string on incorrect parameter given.
```